packages/forms-web-app/src/lib/dashboard-functions.js

```diff
@@ -195,6 +209,56 @@ const getAppealType = (appealCaseData) => {
 	return `${appealCaseData.appealTypeName} appeal`;
 };
 
+/**
+ * @param {AppealCaseWithAppellant} appealCaseData
+ * @returns {boolean}
+ */
+const isEligibilityCompleted = (appealCaseData) => {
+	if (appealCaseData?.appeal?.state === APPEAL_STATE.DRAFT) {
+		const eligibility = appealCaseData.appeal.eligibility;
+		return (
+			eligibility !== undefined &&
+			eligibility.enforcementNotice !== null &&
+			eligibility.applicationDecision !== null &&
+			eligibility.applicationCategories !== null
+		);
+	}
+	return true;
+};
+
+/**
+ * @param {AppealCaseWithAppellant} appealCaseData
+ * @returns {boolean}
+ */
+const hasDecisionDate = (appealCaseData) => {
+	if (appealCaseData?.appeal?.state === APPEAL_STATE.DRAFT) {
+		return appealCaseData.appeal.decisionDate !== null;
+	}
+	return appealCaseData.caseDecisionDate !== null;
+};
+
+/**
+ * @param {AppealCaseWithAppellant} appealCaseData
+ * @returns {boolean}
+ */
+const hasFutureDueDate = (appealCaseData) => {
+	const currentDate = new Date();
+	return (
+		[
+			calculateAppealDueDeadline(
+				appealCaseData.appealTypeCode,
+				appealCaseData.originalCaseDecisionDate
+			),
```

Comment:
In theory an appeal could be completed before the deadline to submit it has passed. So we should branch here: if draft, only include appeal due date, else include the other three.

Reply:
Have refactored calculate appeal due deadline to use existing shared function, and now only calculate appeal due deadline in respect of an AppealSubmission